core/storage/btree.rs

```diff
@@ -3846,32 +3846,31 @@ impl BTreeCursor {
 
     /// Find the index of the cell in the page that contains the given rowid.
     fn find_cell(&mut self, page: &PageContent, key: &BTreeKey) -> Result<CursorResult<usize>> {
-        if self.find_cell_state.0.is_none() {
-            self.find_cell_state.set(0);
-        }
         let cell_count = page.cell_count();
-        while self.find_cell_state.get_cell_idx() < cell_count as isize {
-            assert!(self.find_cell_state.get_cell_idx() >= 0);
-            let cell_idx = self.find_cell_state.get_cell_idx() as usize;
-            match page
-                .cell_get(
-                    cell_idx,
-                    payload_overflow_threshold_max(page.page_type(), self.usable_space() as u16),
-                    payload_overflow_threshold_min(page.page_type(), self.usable_space() as u16),
-                    self.usable_space(),
-                )
-                .unwrap()
-            {
-                BTreeCell::TableLeafCell(cell) => {
-                    if key.to_rowid() <= cell._rowid {
-                        break;
-                    }
-                }
-                BTreeCell::TableInteriorCell(cell) => {
-                    if key.to_rowid() <= cell._rowid {
-                        break;
-                    }
-                }
+        let mut low = 0;
+        let mut high = if cell_count > 0 { cell_count - 1 } else { 0 };
+        let mut result_index = cell_count;
+        if self.find_cell_state.0.is_some() {
+            low = self.find_cell_state.get_cell_idx() as usize;
```

Comment:
Why we don't set `high` value in the state? In case of IO in the middle (possible only for index b-tree with non-u64 key I guess) - we can end up with infinite loop as low will never be changed in the state (because actual key is in the beginning) and `high` is always set to the max value....

Reply:
you are right, i have to store all binary search state
for some reason i convinced myself that setting low would be enough :)